lib/auth/grpcserver.go

```diff
@@ -1723,13 +1723,26 @@ func (g *GRPCServer) DeleteMFADevice(stream proto.AuthService_DeleteMFADeviceSer
 	// Find the device and delete it from backend.
 	devs, err := auth.GetMFADevices(ctx, user)
 	if err != nil {
-		return trace.Wrap(err)
+		return trail.ToGRPC(err)
+	}
+	authPref, err := auth.GetAuthPreference()
+	if err != nil {
+		return trail.ToGRPC(err)
 	}
 	for _, d := range devs {
 		// Match device by name or ID.
 		if d.Metadata.Name != initReq.DeviceName && d.Id != initReq.DeviceName {
 			continue
 		}
+
+		// Make sure that the user won't be locked out by deleting the last MFA
+		// device. This only applies when the cluster requires MFA.
+		if authPref.GetSecondFactor() != constants.SecondFactorOff &&
+			authPref.GetSecondFactor() != constants.SecondFactorOptional &&
```

Comment:
Would removing a last MFA device cause lock-out with `SecondFactorOTP`?

It might be clearer to introduce a helper function which evaluates whether an MFA device must be/remain registered.

Reply:
Please ignore, I now see OTP devices are also registered as a type of MFA.

Reply:
However, the check should probably be stricter though:
* `SecondFactorOTP` => return error when the last **OTP** device is attempted to be removed;
* `SecondFactorU2F` => return error when the last **U2F** device is attempted to be removed.